Zend/tests/list_001.phpt

```diff
@@ -5,6 +5,8 @@
 
 list($a, list($b)) = array(new stdclass, array(new stdclass));
 var_dump($a, $b);
+[$a, [$b]] = array(new stdclass, array(new stdclass));
+var_dump($a, $b);
```

Comment:
You've added eight or so new error conditions and only one of them is currently covered.

I'm asking about `[42] = ...` specifically because it's something the list() implementation would not necessarily check for (as it already had parser level checks for that).